### PR TITLE
Allow insecure for Charge

### LINKED ISSUE #34: Allow insecure connections to Charge (skip https validation)
**Describe the problem/bug**

I'm trying to connect to an existing lightning + charge node that's running on another box, but the charge port has been forwarded over SSH to the btcpayserver one.

**Your environment**
* Version of BTCPay Server: `1.0.4.4`
* Deployment method: independent containers running btcpayserver, nbxplorer and bitcoin RPC port-forwarded
* Other relevant environment details: in order to get around the https requirement for connecting to charge, I've added a https reverse proxy that's listening on 8080 and `proxy_pass`ing to the charge port (9112) and a self-signed localhost cert

**Logs (if applicable)**
Nothing relevant, just New Block events.

**Setup Parameters**
If you're reporting a deployment issue run `. btcpay-setup.sh -i` and paste your the paremeters by obscuring private information.
btcpay-setup.sh not used

**To Reproduce**
1. Go to Store
2. Click on Lightning BTC Modify
3. Scroll down to Url and enter `type=charge;url=https://localhost8080;api-token=xxxx`
4. See error ` Error while connecting to the API (The SSL connection could not be established, see inner exception.)`

It'd be great if there's an `allowinsecure=true` option for charge as well, in order to skip https validation.

### PR DESCRIPTION
**Pull request by Kukks**
fixes #34

### CODE REVIEW

File: src/BTCPayServer.Lightning.Charge/ChargeClient.cs [hunk 1]
```diff
@@ -49,24 +39,54 @@ public ChargeClient(Uri uri, Network network, HttpClient httpClient)
...
+        public ChargeClient(Uri uri, string cookieFilePath, Network network, HttpClient httpClient = null, bool allowInsecure = false)
         {
             if (uri == null)
                 throw new ArgumentNullException(nameof(uri));
             if (network == null)
                 throw new ArgumentNullException(nameof(network));
             if (cookieFilePath == null)
                 throw new ArgumentNullException(nameof(cookieFilePath));
-            httpClient = httpClient ?? _SharedClient;
+            httpClient = CreateHttpClient(uri, allowInsecure, httpClient);
```
Comment (NicolasDorier) [inline src/BTCPayServer.Lightning.Charge/ChargeClient.cs:51]: should be `httpClient ?? _SharedClient`

Comment (Kukks) [reply src/BTCPayServer.Lightning.Charge/ChargeClient.cs:51]: Yes, it reuses the same httpclient when possible. See https://github.com/btcpayserver/BTCPayServer.Lightning/pull/35/files#diff-187e3bd22467c64f3d1c1103ac220062R60

Comment (NicolasDorier) [reply src/BTCPayServer.Lightning.Charge/ChargeClient.cs:51]: it does not, you don't use `_SharedClient` anywhere.

**Comment by NicolasDorier** [comment]
@nitramiz yes, just a bug kukks need to fix.

File: src/BTCPayServer.Lightning.Charge/ChargeClient.cs (hunk 1 above)
Comment (Kukks) [reply src/BTCPayServer.Lightning.Charge/ChargeClient.cs]: fixed

**Comment by NicolasDorier** [comment]
pushed new package

**Pull request merged**